### PR TITLE
Fixes static check failures in test/e2e_node/*

### PR DESCRIPTION
**Pull request by palnabarun**
**What type of PR is this?**
> /kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Ref: #81657 

**Special notes for your reviewer**:
Errors from the static checker:
```
test/e2e_node/dynamic_kubelet_config_test.go:1195:6: func configMapAPIPath is unused (U1000)
test/e2e_node/environment/conformance.go:146:11: this value of err is never used (SA4006)
test/e2e_node/gke_environment_test.go:89:10: this value of err is never used (SA4006)
test/e2e_node/gke_environment_test.go:95:10: this value of err is never used (SA4006)
test/e2e_node/log_path_test.go:160:20: this value of err is never used (SA4006)
test/e2e_node/pids_test.go:86:6: func enablePodPidsLimitInKubelet is unused (U1000)
test/e2e_node/remote/remote.go:139:21: printf-style function with dynamic format string and no further arguments should use print-style function instead (SA1006)
test/e2e_node/runner/remote/run_remote.go:727:38: oauth2.NoContext is deprecated: Use context.Background() or context.TODO() instead.  (SA1019)
test/e2e_node/util.go:68:5: var perlImage is unused (U1000)

```

**Does this PR introduce a user-facing change?**:
```release-note
None
```

### CODE REVIEW

File: test/e2e_node/environment/conformance.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func dns() error {
 		return printSuccess("Dns Check (Optional): %s", success)
 	}
 
-	kubecmd, err := exec.Command("ps", "aux").CombinedOutput()
+	kubecmd, _ := exec.Command("ps", "aux").CombinedOutput()
```
[the remark this answers is not in the document]
Comment (palnabarun) [reply test/e2e_node/environment/conformance.go]: I agree. This error shouldn't be ignored.

File: test/e2e_node/pids_test.go [hunk 2]
```diff
@@ -82,28 +80,6 @@ func makePodToVerifyPids(baseName string, pidsLimit resource.Quantity) *v1.Pod {
 	return pod
 }
 
-// enablePodPidsLimitInKubelet enables pod pid limit feature for kubelet with a sensible default test limit
-func enablePodPidsLimitInKubelet(f *framework.Framework) *kubeletconfig.KubeletConfiguration {
```
[the remark this answers is not in the document]
Comment (palnabarun) [reply test/e2e_node/pids_test.go]: This was added as part of bce9d5f2043bd86964c9fec80d466e47776071bc. I grepped through the code at that commit. There were no instances that used this function.

[the remark this answers is not in the document]
Comment (palnabarun) [reply test/e2e_node/pids_test.go]: I am reaching out to the committer who wrote that piece for more clarifications.

**Comment by palnabarun** [comment]
@BenTheElder I agree that ignoring the errors is not the right fix. Now, checking for those errors. Please review the changes again.

File: test/e2e_node/pids_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (palnabarun) [reply test/e2e_node/pids_test.go]: Digging further into this, found that the function was intended to configure the pod PID limit in kubelet's configuration. In the reference PR, there was a suggestion to use [tempSetCurrentKubeletConfig](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/util.go#L132) instead. Although `tempSetCurrentKubeletConfig` was used, but the function `enablePodPidsLimitInKubelet` was not removed.

Reference:
1. https://github.com/kubernetes/kubernetes/pull/72076
2. https://github.com/kubernetes/kubernetes/pull/72076#discussion_r245774510

[the remark this answers is not in the document]
Comment (palnabarun) [reply test/e2e_node/pids_test.go]: So, in my opinion the function can be removed.

**Comment by BenTheElder** [comment]
/unassign
/uncc
I'm less familiar with the node e2e tests, someone from https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/OWNERS should probably reivew them
/assign @derekwaynecarr

**Comment by palnabarun** [comment]
@SataQiu assigning you for a LGTM for the change to `hack/.staticcheck_failures` :)

/assign @SataQiu

**Comment by BenTheElder** [comment]
/approve
for hack

**Pull request merged**